CDAP-15440 MySQL plugin enhancements

vladhlinskiy opened the pull request.

mlozbin-cybervisiontech commented on database-commons/src/main/java/io/cdap/plugin/db/ConnectionConfig.java:
```diff
@@ -111,17 +130,51 @@ public Properties getConnectionArguments() {
+   */
+  public String getConnectionArgumentsString() {
+    return (this.connectionArguments != null)
+      ? (!getDBSpecificArgumentsString().isEmpty())
```
too hard to read multiple ternary operators in one expression, please split

mlozbin-cybervisiontech commented on database-commons/src/main/java/io/cdap/plugin/db/ConnectionConfig.java:
```diff
@@ -102,6 +105,22 @@ public static Properties getConnectionArguments(@Nullable String connectionArgum
+      return Collections.emptyList();
+    }
+
+    return Stream.of(initQueriesString.split("(?<=;)")).map(String::trim).collect(Collectors.toList());
```
they usually have 1 method call per line

mlozbin-cybervisiontech replied:
it says in javadoc that string is splitted by ";" character

mlozbin-cybervisiontech commented on mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlAction.java:
```diff
@@ -64,9 +104,38 @@ public String getConnectionString() {
-      builder.put(MysqlConstants.ALLOW_MULTIPLE_QUERIES, String.valueOf(true));
-
+      if (useCompression != null) {
+        builder.put(MysqlConstants.USE_COMPRESSION, String.valueOf(useCompression));
```
is it possible to get rid of duplication in getConnectionString methods?

mlozbin-cybervisiontech commented on database-commons/src/main/java/io/cdap/plugin/db/batch/sink/ETLDBOutputFormat.java:
```diff
@@ -160,6 +161,14 @@ private Connection getConnection(Configuration conf) {
+        try (Statement statement = connection.createStatement()) {
+          statement.execute(query);
+        } catch (SQLException e) {
+          LOG.warn("Exception while executing initialization query '" + query + "'", e);
```
it looks like a critical exception to me. I don't think log is enough

mlozbin-cybervisiontech commented on database-commons/src/main/java/io/cdap/plugin/db/batch/sink/AbstractDBSink.java:
```diff
@@ -327,6 +332,20 @@ private void emitLineage(BatchSinkContext context, List<Schema.Field> fields) {
+      try (Statement statement = connection.createStatement()) {
+        statement.execute(query);
+      } catch (SQLException e) {
+        LOG.warn("Exception while executing initialization query '" + query + "'", e);
```
it looks like a critical exception to me. I don't think log is enough

mlozbin-cybervisiontech commented on database-commons/src/main/java/io/cdap/plugin/db/batch/sink/AbstractDBSink.java:
```diff
@@ -258,15 +262,16 @@ private void validateSchema(Class<? extends Driver> jdbcDriverClass, String tabl
+    try (Connection connection = DriverManager.getConnection(connectionString, dbSinkConfig.getConnectionArguments())) {
+      executeInitQueries(connection, dbSinkConfig.getInitQueriesString());
+      try (ResultSet tables = connection.getMetaData().getTables(null, null, tableName, null)) {
+        Preconditions.checkArgument(tables.next(),
```
validate methods should throw InvalidStageException or InvalidConfigPropertyException

mlozbin-cybervisiontech commented on database-commons/src/main/java/io/cdap/plugin/db/batch/source/AbstractDBSource.java:
```diff
@@ -148,16 +150,29 @@ private Schema loadSchemaFromDB(Class<? extends Driver> driverClass)
+      try (Statement statement = connection.createStatement()) {
+        statement.execute(query);
+      } catch (SQLException e) {
+        LOG.warn("Exception while executing initialization query '" + query + "'", e);
```
it looks like a critical exception to me. I don't think log is enough

mlozbin-cybervisiontech commented on database-commons/src/main/java/io/cdap/plugin/db/batch/source/DataDrivenETLDBInputFormat.java:
```diff
@@ -106,6 +107,14 @@ public Connection getConnection() {
+          try (Statement statement = connection.createStatement()) {
+            statement.execute(query);
+          } catch (SQLException e) {
+            LOG.warn("Exception while executing initialization query '" + query + "'", e);
```
it looks like a critical exception to me. I don't think log is enough

mlozbin-cybervisiontech commented on database-commons/src/main/java/io/cdap/plugin/db/batch/sink/AbstractDBSink.java:
```diff
@@ -341,7 +341,8 @@ private void executeInitQueries(Connection connection, List<String> initQueries)
         statement.execute(query);
       } catch (SQLException e) {
-        LOG.warn("Exception while executing initialization query '" + query + "'", e);
+        LOG.error("Exception while executing initialization query '" + query + "'", e);
```
logging error + throwing it is a bad practice. The best way is to throw original exception if possible (without wrapping)

mlozbin-cybervisiontech commented on database-commons/src/main/java/io/cdap/plugin/db/batch/sink/ETLDBOutputFormat.java:
```diff
@@ -166,7 +166,8 @@ private Connection getConnection(Configuration conf) {
           statement.execute(query);
         } catch (SQLException e) {
-          LOG.warn("Exception while executing initialization query '" + query + "'", e);
+          LOG.error("Exception while executing initialization query '" + query + "'", e);
```
logging error + throwing it is a bad practice. The best way is to throw original exception if possible (without wrapping)

mlozbin-cybervisiontech commented on database-commons/src/main/java/io/cdap/plugin/db/batch/source/AbstractDBSource.java:
```diff
@@ -168,7 +169,8 @@ private void executeInitQueries(Connection connection, List<String> initQueries)
         statement.execute(query);
       } catch (SQLException e) {
-        LOG.warn("Exception while executing initialization query '" + query + "'", e);
+        LOG.error("Exception while executing initialization query '" + query + "'", e);
```
logging error + throwing it is a bad practice. The best way is to throw original exception if possible (without wrapping)

mlozbin-cybervisiontech commented on database-commons/src/main/java/io/cdap/plugin/db/batch/source/DataDrivenETLDBInputFormat.java:
```diff
@@ -112,7 +112,8 @@ public Connection getConnection() {
             statement.execute(query);
           } catch (SQLException e) {
-            LOG.warn("Exception while executing initialization query '" + query + "'", e);
+            LOG.error("Exception while executing initialization query '" + query + "'", e);
```
logging error + throwing it is a bad practice. The best way is to throw original exception if possible (without wrapping)

mlozbin-cybervisiontech commented on mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlPostAction.java:
```diff
@@ -47,94 +46,69 @@ public MysqlPostAction(MysqlQueryActionConfig mysqlQueryActionConfig) {
-      }
-
-      return builder.build();
+      return MysqlUtil.composeImmutableDbSpecificArgumentsMap(autoReconnect, useCompression, sqlMode, useSSL,
```
you may omit immutable in the name

mlozbin-cybervisiontech commented on mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlSink.java:
```diff
@@ -31,7 +30,7 @@
  */
 @Plugin(type = BatchSink.PLUGIN_TYPE)
-@Name(MysqlConstants.PLUGIN_NAME)
+@Name(MysqlUtil.PLUGIN_NAME)
```
why did you have to move all the property names to utility class? Lets keep them in config class. I am talking only about names for configuration properties.

albertshau committed:
Set default number of splits to 1

Set the default number of splits for every source plugin to 1.
Without this, the get schema button will fail with a very
confusing message about the query needing a $CONDITIONS clause.

albertshau committed:
Merge pull request #38 from data-integrations/default-num-splits

Set default number of splits to 1

albertshau committed:
bump version to 1.3.0-SNAPSHOT

albertshau committed:
Merge pull request #40 from data-integrations/bump-version-1-3-0-snapshot

bump version to 1.3.0-SNAPSHOT

vhlinskyi committed:
CDAP-15440 Update MySQL configuration to look similar to native
